llama_stack/cli/stack/run.py

```diff
@@ -131,20 +129,48 @@ def _run_stack_run_cmd(self, args: argparse.Namespace) -> None:
         except AttributeError as e:
             self.parser.error(f"failed to parse config file '{config_file}':\n {e}")
 
-        run_args = formulate_run_args(args.image_type, args.image_name, config, template_name)
-
-        run_args.extend([str(config_file), str(args.port)])
-        if args.disable_ipv6:
-            run_args.append("--disable-ipv6")
-
-        for env_var in args.env:
-            if "=" not in env_var:
-                self.parser.error(f"Environment variable '{env_var}' must be in KEY=VALUE format")
-            key, value = env_var.split("=", 1)  # split on first = only
-            if not key:
-                self.parser.error(f"Environment variable '{env_var}' has empty key")
-            run_args.extend(["--env", f"{key}={value}"])
-
-        if args.tls_keyfile and args.tls_certfile:
-            run_args.extend(["--tls-keyfile", args.tls_keyfile, "--tls-certfile", args.tls_certfile])
-        run_with_pty(run_args)
+        # If neither image type nor image name is provided, assume the server should be run directly
+        # using the current environment packages.
+        if not args.image_type and not args.image_name:
+            logger.info("No image type or image name provided. Assuming environment packages.")
+
+            try:
+                from llama_stack.distribution.server.server import main as server_main
+            except ImportError as e:
+                self.parser.error(f"Failed to import server module: {e}")
+
+            try:
+                # Build the server args from the current args passed to the CLI
+                server_args = argparse.Namespace()
+                for arg in vars(args):
+                    # if this is a function, avoid passing it
+                    if callable(getattr(args, arg)):
+                        continue
+                    setattr(server_args, arg, getattr(args, arg))
+
+                # Run the server
+                server_main(server_args)
+            except Exception as e:
+                self.parser.error(f"Failed to run server: {e}")
```

Comment:
again, not a parser error at this point. we are way beyond the point of things being the responsibility of the argparser